internal/rootcoord/meta_table.go

```diff
@@ -936,6 +936,54 @@ func (mt *MetaTable) DropIndex(collName, fieldName, indexName string) (typeutil.
 	return dropIdxID, true, nil
 }
 
+func (mt *MetaTable) GetInitBuildIDs(collName, indexName string) ([]UniqueID, error) {
+	mt.ddLock.RLock()
+	defer mt.ddLock.RUnlock()
+
+	collID, ok := mt.collName2ID[collName]
+	if !ok {
+		collID, ok = mt.collAlias2ID[collName]
+		if !ok {
+			return nil, fmt.Errorf("collection not exist with name = %s", collName)
+		}
+	}
+	collMeta, ok := mt.collID2Meta[collID]
+	if !ok {
+		return nil, fmt.Errorf("collection not exist with name = %s", collName)
+	}
+
+	var indexID typeutil.UniqueID
+	for _, info := range collMeta.FieldIndexes {
+		idxMeta, ok := mt.indexID2Meta[info.IndexID]
+		if !ok {
+			continue
+		}
+		if idxMeta.IndexName != indexName {
+			continue
+		}
+		indexID = info.IndexID
+		break
+	}
+
+	if indexID == 0 {
+		log.Warn("get init buildIDs, index not found", zap.String("collection name", collName),
+			zap.String("index name", indexName))
+		return nil, fmt.Errorf("index not found with name = %s", indexName)
+	}
+
+	initBuildIDs := make([]UniqueID, 0)
+	for _, indexID2Info := range mt.segID2IndexMeta {
+		segIndexInfo, ok := indexID2Info[indexID]
+		if !ok {
+			continue
+		}
+		if !segIndexInfo.ByFlush {
+			initBuildIDs = append(initBuildIDs, segIndexInfo.BuildID)
+		}
+	}
+	return initBuildIDs, nil
```

Comment:
Why we don't return all build ids when `create_index` was called? This is also asynchronouns.

Reply:
If sdk is down, user can't know what buildIDs there are.